games-arcade/lbreakout2_levels/lbreakout2_levels-1.0.recipe

```diff
@@ -0,0 +1,79 @@
+SUMMARY="Additional levels for LBreakout2"
+DESCRIPTION="This package provides 22 additional levels created by users of \
+LBreakout2."
+HOMEPAGE="http://lgames.sourceforge.net/LBreakout2/levelsets.php"
+COPYRIGHT="2010-2018 LBreakout2 users"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI_0="http://lgames.sourceforge.net/LBreakout2/levels/Africa#noarchive"
```

Comment:
Oops, I don't know why, but I though haikuporter would accept things like `SOURCE_URI_AllWorld` and `CHECKSUM_SHA256_AllWorld`. Unfortunately it does not. So the variant I wanted to suggest does not work the way I had expected it to work.
It can be made to work, with an ugly change, but I don't know if this would be an improvement. Well, here it is, with the ugly change, for the curious contributors :)
```
REVISION="1"

levels="Africa AllWorld America Asia Astra Astro Belief Europe Football Frozen Germany \
	HartzIV Hieroglyphs Horror JustForFun Lattices Madness Memory \
	NewBricks Oceania PCGames WallBricks"

SHA256_Africa="0689660047d83a20bd0d61a089af772efb675542d8f059eeb0259022fd4a9624"
SHA256_AllWorld="b3a99e4dcc21582f978568d2e79179512b167e1b507d7c14a168614c23a7a418"
SHA256_America="96d259174bdadc5ddc3e8f000c4570a47e5f54dc330bf5e492228a8d34e8b4d4"
SHA256_Asia="cdf9b0b3be1d050828210f91a573081c12e4274a153ae571da639be923e7a4dc"
SHA256_Astra="a8c35a89e556dc217d82809fd3e4c2c57a17635787067bf1a2798989a7a5cd02"
SHA256_Astro="e4c8063a5539f37668d3c656103f2caf26c9ca61691e332a73c14e8f10e7fe78"
SHA256_Belief="8c1f8aff33543d4170b0b444a878bbe8a7b0ed43a9a9d65384d95cf56ab05595"
SHA256_Europe="4eb0b3dfa89a2df20528ce57de519dd13e74db5e77834d6ca12490820dc7a6ac"
SHA256_Football="764918118353a3925f9dc4144d902f6f7a978c48f76cd5dc892f48cd38e149c6"
SHA256_Frozen="484e020aeb2b0daff06568a59bc8b7beb081c280a57461c83f8309dc1dfec156"
SHA256_Germany="6c36baa112228d7121a319ea717f49277dbd76f2454cf9e40e9a4d6bca591c6d"
SHA256_HartzIV="1079e5385ff4274af4e3ff3f58e8add5f015785ddfc0d444a1eaabe00e434190"
SHA256_Hieroglyphs="a6d7610e9f9b78c3892baf42166dd45d0dfa8c0b97fd08b414db4eb53fb16c9f"
SHA256_Horror="e1b7a032498181d9d7b4ccdb76d819ca218db0a56b703b20bddb8f2553ec944e"
SHA256_JustForFun="0ed257e1264ca9558e599813b1a1332b5b0e013b5357f12e1d290128583e71ff"
SHA256_Lattices="0221870d632669f39ae87df3a74d4eccbb576304a6998ccc9e2d207cfe40ec8c"
SHA256_Madness="df095d66a941902b79f2bd918ff7720c1fa28350b4f63a71ab37bdf19a9a9b47"
SHA256_Memory="a3818ddea66ff6a363b222a85352ce2ce8b562f0ffa9d12e58ebdc5ceca456c9"
SHA256_NewBricks="938c89dad99d30ccc7228e3027fcb7b2b178dd112a7469ec8bdb349496a9e2a3"
SHA256_Oceania="e9e7665104d3703c966708d41f33fe17a7e9ef6ad16008dfccfdd3c485082fa2"
SHA256_PCGames="24e4665e3e89b612e4d2fd38a3bdacf5b55753170c5eb4eac0a82d1f38581431"
SHA256_WallBricks="03d64f974b5d55ef955d5dbedd7179bac31981985ed1ff5de0fb506a89be896e"

i=0
for l in $levels; do
	eval "SOURCE_URI_$i=\"http://lgames.sourceforge.net/LBreakout2/levels/$l#noarchive\""
	eval "CHECKSUM_SHA256_$i=\"\$SHA256_$l\""
	eval "SOURCE_DIR_$i="
	i="`expr $i + 1`"
done

SOURCE_DIR=""
```
```
INSTALL()
{
	mkdir -p "$dataDir"/lbreakout2/levels
	i=0
	for l in $levels; do
		eval f="\$sourceDir$i/$l"
		cp -a "$f" "$dataDir"/lbreakout2/levels
		i="`expr $i + 1`"
	done
}
```

Reply:
Sorry, my `INSTALL` variant was wrong.
```diff
 INSTALL()
 {
 	mkdir -p "$dataDir"/lbreakout2/levels
 	i=0
 	for l in $levels; do
+		if [ $i -eq 1 ]; then
+			f="$sourceDir/$l"
+		else
 			eval f="\$sourceDir$i/$l"
+		fi
 		cp -a "$f" "$dataDir"/lbreakout2/levels
 		i="`expr $i + 1`"
 	done
 }
```
That's because `sourceDir1` is not defined. What gets defined (for `SOURCE_URI_1`) is not `sourceDir1` but `sourceDir`.

Reply:
Let's forget about my ugly variant :)
Your original recipe is better than my variant, especially if the set of available levels does not change too much.